insta/src/App.js

```diff
@@ -0,0 +1,53 @@
+import React, { Component } from 'react';
+import './App.css';
+import dummyData from './dummy-data'
+import SearchBar from "./components/SearchBar/searchBar";
+import PostContainer from './components/PostContainer/PostContainer'
+
+
+class App extends Component {
+  constructor(){
+    super();
+    this.state ={
+      posts: [],
+      searchReturn: [],
+      searchInput: ''
+    }
+  }
+
+ 
+
+  componentDidMount = () =>{
+    this.setState({
+      posts: dummyData,
+      searchReturn: dummyData
+    })
+  }
+ changeHandler=(e)=>{
+// e.preventDefault();
+  this.setState({searchInput: e.target.value},this.searchPostforUser()
+  );}
+
+  searchPostforUser=()=>{
+   this.setState({searchReturn: this.state.posts.filter(item =>{return item.username.toLowerCase().includes(this.state.searchInput.toLocaleLowerCase())}), searchInput: ''})
+  }
+
+  render() {
+    return (
+      <div className="App">
+        <SearchBar change={this.changeHandler} value={this.state.searchInput}/>
+        <div className="post-container">
+          {this.state.searchInput.length?this.state.searchReturn.map(item =>{
+            return(
+            <PostContainer key={item.timestamp} obj={item} />
+            )}):this.state.posts.map(item =>{
+            return(
+            <PostContainer key={item.timestamp} obj={item} />
```

Comment:
On line 42 and 45, you are setting the prop types to the value of obj. This is a very generic naming. It is better to use more descriptive prop names, so as you pass down data it is evident what it is. A better naming would be userData or dummyData